### PR TITLE
[ENHANCEMENT]: `argilla-server`: allow update distribution for non annotated datasets

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR changes the current validator when updating the distribution task to allow updating the distribution task settings for datasets with records without ANY response.

cc @nataliaElv 

**Type of change**

- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/src/argilla_server/models/database.py [hunk 1]
```diff
@@ -359,6 +359,13 @@ class Dataset(DatabaseModel):
         order_by=VectorSettings.inserted_at.asc(),
     )
 
+    responses_count: Mapped[int] = column_property(
+        select(func.count(Response.id))
+        .where(Response.record_id.in_(select(Record.id).where(text("datasets.id == records.dataset_id"))))
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/models/database.py]: Maybe responses in draft status could be filtered for this count (or for the validation)

Comment (jfcalvo) [reply argilla-server/src/argilla_server/models/database.py]: Is not a join of records and responses a better (more efficient) way of doing this count?

Comment (jfcalvo) [reply argilla-server/src/argilla_server/models/database.py]: Btw I think we should not use associations for counting things. I think a helper function could better fit this.

**Review by jfcalvo — APPROVED** [review]

**Comment by frascuchon** [comment]
Fixes https://github.com/argilla-io/argilla/issues/5050

**Pull request merged**